client/my-sites/sharing/buttons/preview-action.jsx

```diff
@@ -1,52 +1,44 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classNames = require( 'classnames' );
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import { omit, startsWith, endsWith } from 'lodash';
 
-module.exports = React.createClass( {
-	displayName: 'SharingButtonsPreviewAction',
+const SharingButtonsPreviewAction = ( props ) => {
+	const { active, position, icon, children } = props;
+	const classes = classNames( 'sharing-buttons-preview-action', {
+		'is-active': active,
+		'is-top': startsWith( position, 'top' ),
+		'is-right': endsWith( position, '-right' ),
+		'is-bottom': startsWith( position, 'bottom' ),
+		'is-left': endsWith( position, '-left' )
+	} );
 
-	propTypes: {
-		active: React.PropTypes.bool,
-		position: React.PropTypes.oneOf( [
-			'top-left',
-			'top-right',
-			'bottom-left',
-			'bottom-right'
-		] ),
-		icon: React.PropTypes.string,
-		onClick: React.PropTypes.func
-	},
+	return (
+		<button type="button" className={ classes } { ...omit( props, [ 'active', 'position' ] ) }>
```

Comment:
Oh, good point.

Another question then: the reason why I suggested enumerating them was that I saw some props listed in the `propTypes` and thought that that's all the props this component can get. I wonder: is it a good idea to list all the possible props that this component can receive in the `propTypes` so other programmers know what they can send or do we just assume that as we render `button`, people would know they can send anything that `button` can accept?